### PR TITLE
Implment OnRecordsSent and delete resend_records within it

### PR DESCRIPTION
**Pull request by darkdh**
fix https://github.com/brave/brave-browser/issues/6834

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by darkdh** [comment]
DEPS will be updated when https://github.com/brave/sync/pull/358 is merged

File: components/brave_sync/brave_profile_sync_service_impl.h [hunk 1]
```diff
@@ -32,6 +32,7 @@ FORWARD_DECLARE_TEST(BraveSyncServiceTest, ClientOnGetInitData);
 FORWARD_DECLARE_TEST(BraveSyncServiceTest, OnGetInitData);
 FORWARD_DECLARE_TEST(BraveSyncServiceTest, OnSaveBookmarksBaseOrder);
 FORWARD_DECLARE_TEST(BraveSyncServiceTest, OnCompactComplete);
+FORWARD_DECLARE_TEST(BraveSyncServiceTest, OnRecordsSent);
```
Comment (AlexeyBarabash) [inline components/brave_sync/brave_profile_sync_service_impl.h]: BraveSyncServiceTest doesn't contain the test named `OnRecordsSent`, looks the actial test is done in `(BraveSyncServiceTest, ExponentialResend)`
So this line is  not required.

File: components/brave_sync/brave_profile_sync_service_impl.h [hunk 2]
```diff
@@ -139,6 +143,7 @@ class BraveProfileSyncServiceImpl
   FRIEND_TEST_ALL_PREFIXES(::BraveSyncServiceTest, ClientOnGetInitData);
   FRIEND_TEST_ALL_PREFIXES(::BraveSyncServiceTest, OnSaveBookmarksBaseOrder);
   FRIEND_TEST_ALL_PREFIXES(::BraveSyncServiceTest, OnCompactComplete);
+  FRIEND_TEST_ALL_PREFIXES(::BraveSyncServiceTest, OnRecordsSent);
```
Comment (AlexeyBarabash) [inline components/brave_sync/brave_profile_sync_service_impl.h]: The same as above.

File: components/brave_sync/brave_profile_sync_service_impl.h (hunk 1 above)
Comment (darkdh) [reply components/brave_sync/brave_profile_sync_service_impl.h]: addressed in 73b682f8175917b4bd8e2356088acb0af9924217

File: components/brave_sync/brave_profile_sync_service_impl.h (hunk 2 above)
Comment (darkdh) [reply components/brave_sync/brave_profile_sync_service_impl.h]: addressed in 73b682f8175917b4bd8e2356088acb0af9924217

**Comment by darkdh** [comment]
all passed except Android
https://ci.brave.com/job/brave-browser-build-pr/job/on-sync-records-sent/6/flowGraphTable/
Going to manually rebase brave-browser and build Android only again

**Pull request merged**